### DISCUSSION TITLE
Issue 8 of POSIX.1-2024

### DISCUSSION THREAD

**Discussion by zzzyxwvut** [vim/vim#15753] [General]
[Issue 8](https://pubs.opengroup.org/onlinepubs/9799919799/) of POSIX.1-2024 has been published on [June 14th](https://www.opengroup.org/austin/).

The applied changes are listed at the very bottom of the `vi`  
[entry](https://pubs.opengroup.org/onlinepubs/9799919799/utilities/vi.html#tag_20_146_25):

> Issue 8
> 
>     Austin Group Defect 251 is applied, encouraging implementations to
>     disallow the creation of filenames containing any bytes that have the
>     encoded value of a <newline> character.
> 
>     Austin Group Defect 1310 is applied, changing the CONSEQUENCES OF ERRORS
>     section.
> 
>     Austin Group Defect 1676 is applied, changing the typeface of some bold
>     text in the RATIONALE section.

For convenience, here are the diffs for the updated parts:

```diff
--- i7
+++ i8
@@ -1,8 +1,6 @@
 CONSEQUENCES OF ERRORS
 
-    When any error is encountered and the standard input is not a terminal device file, vi shall not write the file or return to command or text input mode, and shall terminate with a non-zero exit status.
+    When an unrecoverable error is encountered it shall be equivalent to a SIGHUP asynchronous event.
 
-    Otherwise, when an unrecoverable error is encountered it shall be equivalent to a SIGHUP asynchronous event.
-
-    Otherwise, when an error is encountered, the editor shall behave as specified in Command Descriptions in vi.
+    Otherwise, when an error is encountered, the editor shall behave as specified in Command Descriptions in vi .
 
```

```diff
--- i7
+++ i8
@@ -1,4 +1,4 @@
 FUTURE DIRECTIONS
 
-    None.
+    If this utility is directed to create a new directory entry that contains any bytes that have the encoded value of a <newline> character, implementations are encouraged to treat this as an error. A future version of this standard may require implementations to treat this as an error.

```

The `vi` entries in Issue 7 (2018) and Issue 8 (2024):
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/vi.html
https://pubs.opengroup.org/onlinepubs/9799919799/utilities/vi.html

**Comment by chrisbra** [comment]
Thanks for summarizing those differences. It seems only some marginal changes, probably nothing we need to act on.
